service/src/main/java/tds/assessment/repositories/impl/AssessmentMapper.java

```diff
@@ -0,0 +1,126 @@
+package tds.assessment.repositories.impl;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import tds.assessment.Assessment;
+import tds.assessment.Property;
+import tds.assessment.Segment;
+
+/**
+ * Mapper used to map results from the AssessmentQuery for assessments by key
+ */
+class AssessmentMapper {
```

Comment:
This was pretty complicated so I extracted the code that was in the repository into this separate class.  The mapping no longer requires a specific order to work properly.  The results from the query can have multiple queries for the same segment/assessment.  For example, if there is two different languages associated with the assessment there will be two `rows`.